crates/javy/src/serde/err.rs

```diff
@@ -11,7 +12,13 @@ impl error::Error for Error {}
 impl fmt::Display for Error {
     fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
         match self {
-            Error::Custom(e) => formatter.write_str(&e.to_string()),
+            Error::Custom(e) => {
+                if let Some(e) = e.downcast_ref::<JSError>() {
+                    formatter.write_str(&format!("JSError: {}", &e.to_string()))
```

Comment:
The context is lost when triggering an error from the deserializer and the error is handled by the serializer (serde_json); it only happens in the case of stringify because the serde_json serializer is on the receiving end of the transcoding process. The introduction of `JSError: ....` intentional to be able to match against a known exception let the exception bubble up. See the serde transcode issue linked in https://github.com/bytecodealliance/javy/pull/662#discussion_r1633726753